muskathlon/forms/muskathlon_registration_form.py

```diff
@@ -207,9 +207,11 @@ def form_before_create_or_update(self, values, extra_values):
             if not partner.ambassador_details_id:
                 # Creation of ambassador details reloads cache and remove
                 # all field values. We therefore run it in a job.
+                sporty = self.env.ref('partner_compassion.engagement_sport')
                 partner.with_delay().create_ambassador_details({
                     'partner_id': partner.id,
-                    'advocacy_source': 'Online Muskathlon registration'
+                    'advocacy_source': 'Online Muskathlon registration',
+                    'engagement_ids': [sporty.id]
```

Comment:
I think it should be `[(4, sporty.id)]` to add the relation. (see odoo documentation on the write method)